File: Desktop/SummerVacation/0702/src/numTwentyNine.java
import java.util.Scanner;
public class numTwentyNine {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        boolean finish = true;
        while(finish) {
            boolean check = true;
            System.out.print("What is the rate of return? ");
            String input = scan.next();
            char[] result = input.toCharArray();
            for(char c : result){
                if(!Character.isDigit(c))
                    check = false;
            }
            if(check) {
                if(Integer.parseInt(input) != 0){
                System.out.printf("It will take %d years.",(72 / Integer.parseInt(input)));
                finish = false;
                }
                else
                    System.out.printf("%s is invalid input.\n", input );
            }
            else{
                System.out.printf("%s is invalid input.\n", input );
            }
        }

    }

}

File: Desktop/SummerVacation/0702/python/numTwentyNine.py
while True :
    answer = input("What is the rate of return? ")
    if answer.isdigit() and int(answer) != 0 :
        break;
    else :
        print("Sorry. That's not a valid input")
result = 72 / int(answer)
print("It will take %d years to double your inital investment." % result)